controllers/configuration_controller_test.go

```diff
@@ -685,6 +699,114 @@ configuration:
 	}
 }
 
+func TestConfigurationShouldReconcile(t *testing.T) {
+	testcase := []struct {
+		name             string
+		errStr           string
+		snapshot         func(name, namespace string) *v1alpha1.Snapshot
+		componentVersion func() *v1alpha1.ComponentVersion
+	}{
+		{
+			name: "should not reconcile in case of matching generation and existing snapshot with ready state",
+			componentVersion: func() *v1alpha1.ComponentVersion {
+				cv := DefaultComponent.DeepCopy()
+				cv.Status.ObservedGeneration = 99
+				return cv
+			},
+			snapshot: func(name, namespace string) *v1alpha1.Snapshot {
+				snapshot := &v1alpha1.Snapshot{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: namespace,
+					},
+					Spec:   v1alpha1.SnapshotSpec{},
+					Status: v1alpha1.SnapshotStatus{},
+				}
+				conditions.MarkTrue(snapshot, meta.ReadyCondition, meta.SucceededReason, "Snapshot with name '%s' is ready", snapshot.Name)
+				return snapshot
+			},
+		},
+		{
+			name:   "should reconcile if snapshot is not ready",
+			errStr: "failed to reconcile mutation object: failed to fetch resource data from resource ref: failed to fetch resource from resource ref: unexpected number of calls; not enough return values have been configured; call count 0",
```

Comment:
This error verifies that the fake was called. Therefore, it proceeded with the reconciliation.